dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java

```diff
@@ -427,14 +441,30 @@ private void processContentDependency(final Contentlet contentlet)
             final List<Contentlet> contentList =
                     APILocator.getContentletAPI().findAllVersions(ident, false, user, false);
 
-            final Set<Contentlet> contentsToProcess = new HashSet<Contentlet>();
-            final Set<Contentlet> contentsWithDependenciesToProcess = new HashSet<Contentlet>();
+            final Set<Contentlet> contentsToProcess = new HashSet<>();
+            final Set<Contentlet> contentsWithDependenciesToProcess = new HashSet<>();
 
             for (final Contentlet contentletVersion : contentList) {
 
                 if (contentletVersion.isHTMLPage()) {
                     processHTMLPagesDependency(contentletVersion.getIdentifier());
                 }
+                // Contentlet Dependencies in Story Block fields, if applicable
+                if (contentlet.getContentType().hasStoryBlockFields()) {
+                    this.storyBlockAPI.get().getDependencies(contentletVersion).forEach(contentletId -> {
+                        try {
+                            final Contentlet contentInStoryBlock =
+                                    this.contentletAPI.get().findContentletByIdentifier(contentletId,
+                                            contentletVersion.isLive(), contentletVersion.getLanguageId(),
+                                            APILocator.systemUser(), false);
+                            tryToAddAndProcessDependencies(PusheableAsset.CONTENTLET, contentInStoryBlock,
+                                    ManifestReason.INCLUDE_DEPENDENCY_FROM.getMessage(contentletVersion));
+                        } catch (final DotDataException | DotSecurityException e) {
+                            Logger.warn(this,
+                                    String.format("Could not analyze dependent Contentlet '%s' referenced " + "in " + "Story Block field from Contentlet Inode " + "'%s': %s", contentletId, contentletVersion.getInode(), e.getMessage()));
+                        }
+                    });
```

Comment:
maybe we can move this to a local method something like: processStoryBlock(...)

Reply:
yeah

Reply:
Done!